### ISSUE TITLE
Linux or MAC OSX server

### ISSUE THREAD

**Issue by felipk** [phoboslab/jsmpeg-vnc#5]
Hi,

Its possible to use a linux or Mac OS as a server?

If I compile the source code it will work?

Thanks

**Comment by phoboslab** [comment]
No. The [grabber.c](https://github.com/phoboslab/jsmpeg-vnc/blob/master/source/grabber.c) code to get pixel data from a window is quite Windows specific. You'd need to rewrite this for other OSes. The [app_on_message()](https://github.com/phoboslab/jsmpeg-vnc/blob/master/source/app.c#L135) that "fakes" keyboard and mouse input, only works on Windows as well.

Everything else, i.e. the encoder and WebSocket server, should work cross platform.

**Comment by jaredly** [comment]
Could you add something to the readme (or the GH summary line) saying this is Windows-only? That would be helpful.

**Comment by phoboslab** [comment]
@jaredly good idea. The summary now reads: _A low latency, high framerate screen sharing server for Windows and client for browsers_

**Comment by jaredly** [comment]
awesome thanks!

**Comment by n1trux** [comment]
Could the screen grabbing on Linux be possible with xlib? https://tronche.com/gui/x/xlib/

**Comment by g4jc** [comment]
I would also be very interested in GNU/Linux support! This is a very nice program.

**Comment by Xeddius** [comment]
I would love GNU/Linux and Mac support!
ffmpeg can already grab the screen natively on linux. So that solves the screen-grabbing issues.
Which just leaves support for the input emulation, which I believe xdotool can solve!

**Comment by alexsarmiento** [comment]
WinPR (Windows Portable Runtime) by FreeRDP should help to solve this kind of portability issues.

**Comment by thesauri** [comment]
In case this is still relevant to anyone, I did a semi-complete port to Linux a couple of years ago that should still work: https://github.com/thesauri/jsmpeg-vnc-linux
